app/server/Main.hs

```diff
@@ -1,6 +1,79 @@
+{-# LANGUAGE QuasiQuotes #-}
+
+{-
+   This module parses command line arguments and starts the wallet.
+
+   This should be the only module that has a notion of command line arguments.
+   Although, for development purposes, we may extend the docopt specification
+   here and call `getArgs` where it is needed.
+-}
 module Main where
 
+import Control.Monad
+    ( when )
 import Prelude
+import System.Console.Docopt
+    ( Docopt
+    , docopt
+    , exitWithUsage
+    , getArgOrExitWith
+    , isPresent
+    , longOption
+    , parseArgsOrExit
+    )
+import System.Environment
+    ( getArgs )
+import Text.Read
+    ( readMaybe )
+
+
+-- | Command-Line Interface specification. See http://docopt.org/
+cli :: Docopt
+cli = [docopt|
+cardano-wallet-server
+
+Start the cardano wallet along with its API and underlying node.
+
+Usage:
+  cardano-wallet-server [options]
+  cardano-wallet-server --help
+
+Options:
+  --wallet-server-port <PORT>  port used for serving the wallet API [default: 8090]
+  --node-port <PORT>           port used for node-wallet communication [default: 8080]
+  --network <NETWORK>          mainnet or testnet [default: mainnet]
+|]
+
+
+data Network = Mainnet | Testnet
+    deriving (Show, Enum)
 
 main :: IO ()
-main = return ()
+main = do
+    args <- parseArgsOrExit cli =<< getArgs
+    when (args `isPresent` (longOption "help")) $ exitWithUsage cli
+
+    let get option = args `getArgOrExit` option
+
+    network <- readNetwork =<< get (longOption "network")
+    walletPort <- readInt <$> get (longOption "wallet-server-port")
+    nodePort <- readInt <$> get (longOption "node-port")
+
+    putStrLn $
+        "TODO: start wallet on port " ++ (show walletPort) ++
+        ",\n      connecting to " ++ (show network) ++
+        " node on port " ++ (show nodePort)
+  where
+    readInt :: String -> Int
+    readInt s = maybe (error $ "not an integer: " ++ show s) id (readMaybe s)
```

Comment:
I guess we could do something more proper and testable for this, by having everything in
```haskell
:: [String] -> Either ErrorOutput a
```
or something